test/e2e/proxy_test.go

```diff
@@ -188,3 +212,127 @@ func createPreexistingUserAndIdentity(t *testing.T, user proxyUsers) (*userv1.Us
 	require.NoError(t, user.expectedMemberCluster.CreateWithCleanup(context.TODO(), preexistingIdentity))
 	return preexistingUser, preexistingIdentity
 }
+
+func newWsWatcher(t *testing.T, user proxyUser, proxyURL string) *wsWatcher {
+	u, err := url.Parse(proxyURL)
+	require.NoError(t, err)
+	return &wsWatcher{
+		t:         t,
+		user:      user,
+		proxyHost: u.Host,
+	}
+}
+
+// wsWatcher represents a watcher which leverages a WebSocket connection to watch for config maps in the user's namespace.
+// The connection is established with the reg-service proxy instead of direct connection to the API server.
+type wsWatcher struct {
+	done       chan interface{}
+	interrupt  chan os.Signal
+	t          *testing.T
+	user       proxyUser
+	connection *websocket.Conn
+	proxyHost  string
+
+	mu          sync.RWMutex
+	receivedCMs map[string]*corev1.ConfigMap
+}
+
+// start creates a new WebSocket connection. The method returns a function which is to be used to close the connection when done.
+func (w *wsWatcher) Start() func() {
```

Comment:
this method doesn't need to be exported

Reply:
That's right but I decided to export `Start()` and `WaitForConfigMap()` as the only "public" API of the watcher struct. Only those two methods are supposed to be used by clients. Other methods only to be used by the watcher itself.
So if you don't mind I would keep it as is in attempt to make it more clear ("external" vs. internal API).